src/sdk/service/network-validation-service/__tests__/network-validation-service.test.unit.ts

```diff
@@ -0,0 +1,388 @@
+import { NetworkValidationService } from '../network-validation-service';
+
+const dummyIpOne = '192.0.0.1';
+const dummyIpTwo = '10.0.0.1';
+const dummyIpThree = '192.168.0.1';
+const dummyIpArrayOne = ['10.0.0.0', '10.0.0.1', '10.0.0.2'];
+const dummyIpArrayTwo = ['10.0.0.11', '10.0.0.12', '10.0.0.13'];
+const dummyIpArrayThree = [
+  '10.0.0.1',
+  '10.0.0.2',
+  '10.0.0.3',
+  '10.0.0.4',
+  '10.0.0.5',
+  '10.0.0.6',
+  '10.0.0.7',
+  '10.0.0.8',
+  '10.0.0.9'
+];
+const dummyGateway = '192.30.250.00';
+const dummyNetmaskOne = '255.255.192.0';
+const dummyNetmaskTwo = '255.255.255.0';
+const dummyNetmaskThree = '255.255.255.255';
+const dummyIPv4AddressRangeOne = {
+  start: '10.0.0.1',
+  end: '10.0.0.10'
+};
+const dummyIPv4AddressRangeTwo = {
+  start: '10.0.0.10',
+  end: '10.0.0.255'
+};
+
+test('Check if IP Address is valid and not reserved.', () => {
+  expect(NetworkValidationService.isValidIPv4Address('74.203.180.179', false)).toBeTruthy();
+});
+
+test('Check if IP Address is valid and not reserved.', () => {
+  expect(NetworkValidationService.isValidIPv4Address('74.203.180.255', false)).toBeFalsy();
+});
+
+test('Check if Netmask is valid.', () => {
+  expect(NetworkValidationService.isValidNetmask('255.255.255.128')).toBeTruthy();
+});
+
+test('Check if Netmask is invalid.', () => {
+  expect(NetworkValidationService.isValidNetmask('255.255.0.128')).toBeFalsy();
+});
+
+test('Check if Netmask is invalid.', () => {
```

Comment:
should be combined with the previous one (one test case, multiple assertions). Also, I would expect to see more assertions for both thevalid and invalid cases. i.e. a random string, a netmask with only 3 octets, an empty string, etc. It takes so little time to add an extra assertion that you have no reason to not test things more thoroughly.